api/novelist.py

```diff
@@ -551,31 +569,50 @@ def create_item_object(self, object, currentIdentifier, existingItem):
             # a data error.
             return (None, None, None, False)
 
-        role = object[5]
-        author = object[6] if role in Contributor.AUTHOR_ROLES else ""
+        roles = list(Contributor.AUTHOR_ROLES)
+        roles.append(Contributor.NARRATOR_ROLE)
+
+        role = object[6]
+        author_or_narrator = object[7] if role in roles else ""
+        distributor = object[8]
 
         # If we encounter an existing ISBN and its role is "Primary Author",
         # then that value overrides the existing Author property.
-        #
-        # TODO: add 'narrator' field when we encounter a Narrator role.
+        # If the role is narrator, then we don't need to worry about this step.
```

Comment:
Here, I am assuming that, unlike the author role, there won't be another narrator role to update the value for. Is that correct to assume? (In this one, if "Primary Author" was found after "Author", that value replaced the first one).

Reply:
Yeah, there's no equivalent role to "Primary Narrator". If a book has multiple narrators I think picking the first one is fine.